poppy/poppy_core.py

```diff
@@ -2369,7 +2369,13 @@ def display(self, nrows=1, row=1, what='intensity', crosshairs=True, ax=None, co
 
         if self.pixelscale is not None:
             # TODO handle units better here for pupil vs. image planes? meters/pix vs arcsec/pix
-            halfsize = self.pixelscale.value * self.amplitude.shape[0] / 2
+            if self.pixelscale.decompose().unit ==u.m/u.pix:
```

Comment:
Nice! You can probably remove the "TODO" comment on line 2371, since you have now done it. Thanks!